core/src/main/scala/kafka/utils/CommandLineUtils.scala

```diff
@@ -64,7 +64,7 @@ object CommandLineUtils extends Logging {
 
     val props = new Properties
     for(a <- splits) {
-      if (a.length == 1) props.put(a(0), "")
+      if (a.length == 1) throw new IllegalArgumentException(s"Missing value for key ${a(0)}")
```

Comment:
@ijuma to not introduce any unwanted error or change in cli behavior in a minor release, I am a +1 on adding a boolean param, `acceptEmptyValue`, with a default value of `true` to `acceptEmptyValue`.